UOP1_Project/Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -8,50 +8,42 @@ public class Inventory : ScriptableObject
 {
 	[Tooltip("The collection of items and their quantities.")]
 	[SerializeField]
-	private Dictionary<Item, int> _items = new Dictionary<Item, int>();
-	public IReadOnlyDictionary<Item, int> Items => _items;
-
-	public void Add(Item item, int count = 1)
-	{
-		if (count <= 0)
-			return;
-
-		if (!_items.ContainsKey(item))
-		{
-			_items.Add(item, 0);
-		}
-
-		_items[item] += count;
-	}
-
-	public void Remove(Item item, int count = 1)
-	{
-		if (count <= 0)
-			return;
-
-		if (!_items.ContainsKey(item))
-			return;
-
-		_items[item] -= count;
-
-		if (_items[item] <= 0)
-		{
-			_items.Remove(item);
-		}
-	}
-
-	public bool Contains(Item item)
-	{
-		return _items.ContainsKey(item);
-	}
-
-	public int Count(Item item)
-	{
-		if (!_items.ContainsKey(item))
-		{
-			return 0;
-		}
-
-		return _items[item];
-	}
+	private List<ItemStack> _items = new List<ItemStack>();
+	public List<ItemStack> Items => _items;
+
+    public void Add(Item item, int count = 1)
+    {
+        _items.Add(new ItemStack(item, count));
+    }
+
+    //public void Remove(Item item, int count = 1)
+    //{
+    //    if (count <= 0)
+    //        return;
+
+    //    if (!_items.ContainsKey(item))
+    //        return;
+
+    //    _items[item] -= count;
+
+    //    if (_items[item] <= 0)
+    //    {
+    //        _items.Remove(item);
+    //    }
+    //}
+
+    //public bool Contains(Item item)
+    //{
+    //    return _items.ContainsKey(item);
+    //}
```

Comment:
We need to iterate through the List. If we find an ItemStack where itemStack.Item == item, return true. If we get through the whole list without returning true, return false.